build/.gitignore

```diff
@@ -1 +0,0 @@
-*
```

Comment:
This was here to help the humans. 1: no need to make the dir, 2: doen't show all the build files to git

Reply:
OK, I've added the directory back. The new cmake way with `cmake -S. -Bbuild` creates the directory for you but this is only available in CMake 3.15+

The gitignore is configured for all `build` directories in the top-level `.gitignore`, so I've added it with the more idiomatic `.gitkeep` instead.